libs/navigation/bootstrapper.js

```diff
@@ -35,4 +35,41 @@ export default async function bootstrapBlock(miloLibs, blockConfig) {
       loadPrivacy(getConfig, loadScript);
     }, blockConfig.delay);
   }
+
+  /** Jarvis Chat */
+  const isChatInitialized = (client) => !!client?.isAdobeMessagingClientInitialized();
```

Comment:
I'm thinking whether we want to load the extra code for all consumers on load or maybe delay the loading or even load conditionally.

Reply:
Yes, we should check if jarvis config is passed and then invoke _addDomEvents_. @Deva309

Reply:
Done

Reply:
But still, this is loading (not executing) the code for all the consumers. Do we really need that?

Reply:
Would this mean we extract jarvis code in a separate file and load it conditionally based on client consumption? If yes, wouldn't that cause an extra network call?

Reply:
it would, but only for consumers who load jarvis; we're doing the same thing in the Milo implementation https://github.com/adobecom/milo/blob/stage/libs/scripts/delayed.js#L23

Reply:
it's not a blocker for merging the code though, just a suggestion; if you think it's better with loading the code every time and increasing the codebase, then go ahead and merge, but things might get hard to maintain over time.